From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 4 Feb 2018 18:55:09 +0100
Subject: [PATCH] {For LLVMLinux} crypto/sha3_generic: clang does not known
 optimize attribute

clang reports:

    crypto/sha3_generic.c:38:28: error: unknown attribute '__optimize__'
    ignored [-Werror,-Wunknown-attributes]
    static void __attribute__((__optimize__("O3"))) keccakf(u64 st[25])
                               ^
---
 crypto/sha3_generic.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/crypto/sha3_generic.c b/crypto/sha3_generic.c
index a965b9d80559..b553cc930018 100644
--- a/crypto/sha3_generic.c
+++ b/crypto/sha3_generic.c
@@ -35,7 +35,11 @@ static const u64 keccakf_rndc[24] = {
 
 /* update the state with given number of rounds */
 
-static void __attribute__((__optimize__("O3"))) keccakf(u64 st[25])
+static void
+#ifndef __clang__
+__attribute__((__optimize__("O3")))
+#endif
+keccakf(u64 st[25])
 {
 	u64 t[5], tt, bc[5];
 	int round;
-- 
